certora/vault_properties.md

```diff
@@ -0,0 +1,109 @@
+# Properties of Vault
+
+## Overview of the Vault
+
+The Vault contract is the core of the SandClock system. It allows user to deposit funds to invest and withdraw funds to disinvest. It delegates investing activity to a contract that implements [`IStrategy`](https://github.com/lindy-labs/sc_solidity-contracts/blob/main/contracts/strategy/IStrategy.sol) interface. 
+
+The main value added by Vault is allowing user to specify multiple beneficiaries in a very flexible way, which opens a lot of possibilities. It also allows sponsors to contribute yield to all the users while guaranteeing sponsors to be the last ones to bear investment loss.
+
+It uses `shares` to keep track of users' entitlements to the underlying assets. Whenever user deposits accepted assets into the vault, the Vault will add a certain amount of shares against the user account. When user withdraws underlying assets, their shares are reduced.
+
+Vault and Strategy is one-to-one relationship, i.e., each Vault can have only one strategy and vice versa.
+
+The Vault has the following state variables:
+* `totalSponsored` (type `uint256`), total sponsored amount of the underlying ERC20 asset
+* `totalShares` (type `uint256`), total shares of the all the users
+* `depositGroupIdOwner` (type `mapping(uint256 => address)`), a map from *depositGroupId* to its owner.
```

Comment:
Yes, I deliberately skipped all the private state variables, since they are not exposed to external hence not testable and has no affect on external world as well. But for the sake of readability, I've added a sentence explaining what is *depositGroupId*